haystack/schema.py

```diff
@@ -258,4 +260,21 @@ def run(self, *args: Any, **kwargs: Any):
         :param kwargs:
         :return:
         """
-        pass
\ No newline at end of file
+        pass
+
+    def set_pipeline_config(self, **kwargs):
```

Comment:
I'd suggest to rename this to something independent of pipeline because it could also be used independently for storing a "component representation" or logging params.

Maybe something like `set_config()` or `register_params()`?